src/LibChorusTests/VcsDrivers/Mercurial/HgSettingsTests.cs

```diff
@@ -354,6 +354,30 @@ public void EnsureTheseExtensionAreEnabled_noExistingExtensions_AddsThem()
 		}
 
 		[Test]
+		public void EnsureTheseExtensionAreEnabled_ExtraExtension_RemovesThem()
+		{
+			using(new MercurialIniForTests())
+			using(var testRoot = new TemporaryFolder("ChorusHgSettingsTest"))
+			{
+				HgRepository.CreateRepositoryInExistingDir(testRoot.Path, _progress);
+				var repository = new HgRepository(testRoot.Path, new ConsoleProgress());
+				File.WriteAllText(testRoot.Combine(Path.Combine(".hg", "hgrc")), @"
+[extensions]
+a =
+x =
+");
+				var extensions = new Dictionary<string, string>();
+				extensions.Add("a", "");
+				extensions.Add("b", "");
+				repository.EnsureTheseExtensionsAndFormatSet(extensions);
+
+				Assert.AreEqual(2, repository.GetEnabledExtension().Count());
+				Assert.AreEqual("a", repository.GetEnabledExtension().ToArray()[0]);
+				Assert.AreEqual("b", repository.GetEnabledExtension().ToArray()[1]);
+			}
+		}
```

Comment:
The .hgrc was created with 'a', and 'x' while the dictionary has 'a' and 'b'. The count on the result combined with verifying 'a' and 'b' in the result proves 'x' was removed.
I've taken the opportunity to add appropriate failure messages on the asserts.